build-tools/build.sh

```diff
@@ -0,0 +1,217 @@
+}
+
+function get_library_name() { 
+    library_name=$(awk -F'=' '/^LIBNAME:=/{print $2}' Makefile)
+    echo "library_name=$library_name" >> "$GITHUB_OUTPUT"
+    echo "Library name found: $library_name"
+    echo $library_name >> "library_name.txt"
+}
+
+get_sha &
+get_version &
+get_library_name &
+wait
+
+version=$(cat version.txt)
+rm version.txt
+library_name=$(cat library_name.txt)
+rm library_name.txt
+
+echo "Version before setting postfix: $version"
+echo "SHA before setting postfix: $sha"
+
+if [ "$INPUT_NO_COMMIT_HASH" == "true" ]; then
+    postfix="${version}"
+else
+    postfix="${version}+${sha}"
+fi
+echo "Postfix after setting: $postfix"
+echo "postfix=$postfix" >> "$GITHUB_OUTPUT"
+
+name="$library_name@$postfix"
+echo "name=$name" >> "$GITHUB_OUTPUT"
+echo "Name found: $name"
+
+echo "::endgroup::"
+# ----------------
+# BUILDING PROJECT
+# ----------------
+# Pause errors
+set +e
+pros make clean
+ERR_OUTPUT=$(mktemp)
+STD_OUTPUT=$(mktemp)
+# Set IS_LIBRARY to 0 to build the project if $template is 1
+if (($template == 1)); then
+    echo "::group::Building ${name}"
+    echo "Setting IS_LIBRARY to 0"
+    sed -i "s/^IS_LIBRARY:=.*\$/IS_LIBRARY:=0/" Makefile
+fi
+
+
+if [[ "$INPUT_MULTITHREADING" == "true" ]]; then
+    echo "Multithreading is enabled"
+    start_build_time=$SECONDS
+    make quick -j 2> $ERR_OUTPUT | tee $STD_OUTPUT
+    build_time=$((SECONDS - $start_build_time))
+else
+    echo "Multithreading is disabled"
+    start_build_time=$SECONDS
+    make quick 2> $ERR_OUTPUT | tee $STD_OUTPUT
+    build_time=$((SECONDS - $start_build_time))
+fi
+
+
+if (($template == 1)); then
+    echo "Setting IS_LIBRARY back to 1"
+    sed -i "s/^IS_LIBRARY:=.*\$/IS_LIBRARY:=1/" Makefile
+    echo "::endgroup::"
+fi
+
+STD_EDITED_OUTPUT=$(mktemp)
+# Remove ANSI color codes from the output
+sed -e 's/\x1b\[[0-9;]*m//g' $STD_OUTPUT > $STD_EDITED_OUTPUT
+
+if [ -s "$ERR_OUTPUT" ]; then
+    error_output=$(cat "$ERR_OUTPUT")
+    norm_output=$(cat "$STD_EDITED_OUTPUT")
+    echo "# 🛑 Build Failed" >> $GITHUB_STEP_SUMMARY
+    echo "#### 📄 Error Output" >> $GITHUB_STEP_SUMMARY
+    echo "Build failed in $build_time seconds" >> $GITHUB_STEP_SUMMARY
+    echo "Total Build Script Runtime: $(($SECONDS - $script_start_time)) seconds" >> $GITHUB_STEP_SUMMARY
+    echo "<details><summary>Click to expand</summary>" >> $GITHUB_STEP_SUMMARY
+    echo "" >> $GITHUB_STEP_SUMMARY
+    echo "" >> $GITHUB_STEP_SUMMARY
+    echo "\`\`\`" >> $GITHUB_STEP_SUMMARY
+    echo "$norm_output" >> $GITHUB_STEP_SUMMARY
+    echo "\`\`\`" >> $GITHUB_STEP_SUMMARY
+    echo "</details>" >> $GITHUB_STEP_SUMMARY
+    exit 1
+fi
+
+set -e
+# -----------------
+# CREATING TEMPLATE
+# -----------------
+
+if (($template == 1)); then
+echo "::group::Updating Makefile"
+
+sed -i "s/^VERSION:=.*\$/VERSION:=${postfix}/" Makefile
+
+cat Makefile
+
+echo "::endgroup::"
+
+
+
+echo "::group::Creating ${name} template"
+
+pros make template
```

Comment:
This could be done in parallel with compiling the template, as long as you have a fake `LemLib.a` file.

Reply:
wdym

Reply:
After some research, I have found that this would likely require pros-fake, and is not within the scope of the PR.